src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableHashSet_1.cs

```diff
@@ -754,8 +754,13 @@ private static bool Overlaps(IEnumerable<T> other, MutationInput origin)
         private static bool SetEquals(IEnumerable<T> other, MutationInput origin)
         {
             Requires.NotNull(other, nameof(other));
+            ISet<T> otherSet = other switch
```

Comment:
Do you have any benchmarks showing improvement? This is trading off a `HashSet<T>` allocation with runtime type tests, using a lot of `ISet<T>` virtual calls (and an enumerator allocation when iterating over `otherSet`) so it doesn't seem like a clear cut improvement at face value.